homeassistant/components/homematic/__init__.py

```diff
@@ -262,7 +262,7 @@ def setup(hass, config):
     # Create hosts-dictionary for pyhomematic
     for rname, rconfig in conf[CONF_INTERFACES].items():
         remotes[rname] = {
-            'ip': socket.gethostbyname(rconfig.get(CONF_HOST)),
+            'ip': rconfig.get(CONF_HOST),
```

Comment:
That is only for xmlrpc server they we run. You are sure that you not only mean the settings from hosts part?

Reply:
Sorry I don't get what you mean. I'm using Home Assistant for three days now and not very familiar with the whole stuff. Without the changes I got the following exception:

```
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic.connection] HMConnection: Creating server object
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] ServerThread.__init__
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] __init__: Creating proxies
2019-04-16 07:58:42 INFO (SyncWorker_19) [pyhomematic._hm] Creating proxy rf. Connecting to 192.168.100.74:42001
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Getting local ip
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Got local ip 192.168.100.2
2019-04-16 07:58:42 INFO (SyncWorker_19) [pyhomematic._hm] Creating proxy ip. Connecting to 192.168.100.74:42010
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Getting local ip
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Got local ip 192.168.100.2
2019-04-16 07:58:42 INFO (SyncWorker_19) [pyhomematic._hm] Creating proxy raspberrymatic. Connecting to 192.168.100.74:2001
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Getting local ip
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] LockingServerProxy.__init__: Got local ip 192.168.100.2
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] RPCFunctions.__init__
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] RPCFunctions.__init__: iterating proxy = homeassistant-rf
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] RPCFunctions.__init__: iterating proxy = homeassistant-ip
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] RPCFunctions.__init__: iterating proxy = homeassistant-raspberrymatic
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] ServerThread.__init__: Setting up server
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] ServerThread.__init__: Registering RPC functions
2019-04-16 07:58:42 INFO (Thread-3) [pyhomematic._hm] Starting server at http://0.0.0.0:47051
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] ServerThread.proxyInit: init('http://192.168.100.2:47051', 'homeassistant-rf')
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] proxyInit: Exception: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: IP address mismatch, certificate is not valid for '192.168.100.74'. (_ssl.c:1056)
2019-04-16 07:58:42 WARNING (SyncWorker_19) [pyhomematic._hm] Failed to initialize proxy
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] ServerThread.proxyInit: init('http://192.168.100.2:47051', 'homeassistant-ip')
2019-04-16 07:58:42 DEBUG (SyncWorker_19) [pyhomematic._hm] proxyInit: Exception: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: IP address mismatch, certificate is not valid for '192.168.100.74'. (_ssl.c:1056)
2019-04-16 07:58:42 WARNING (SyncWorker_19) [pyhomematic._hm] Failed to initialize proxy
```

All those created proxies are connecting to an IP address (as previously forced). And as far as I understand the settings used beneath the line [pyhomematic/_hm.py#L521](https://github.com/danielperna84/pyhomematic/blob/8b91f3e84c83f05d289c740d507293a0d6759d8e/pyhomematic/_hm.py#L521) come from ```remotes``` settings.

But maybe I am completely lost at the moment :-/

Reply:
@pvizeli 
`socket.gethostbyname` turns `myhost.local` into `192.168.1.123`. So even with the hostname correctly set in configuration.yaml, the component initializes pyhomematic with the IP addresses. pyhomematic has gained SSL support since the release of the CCU3 (and recent RaspberryMatic releases). Verification of course fails because the IP address never matches the hostname of the certificate.